pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -521,6 +532,52 @@ func (l LibvirtWrapper) SetupLibvirt() (err error) {
 	return nil
 }
 
+func getLibvirtLogFilters() (logFilters string, enableDebugLogs bool) {
+
+	_, envVarExists := os.LookupEnv(services.ENV_VAR_LIBVIRT_DEBUG_LOGS)
+	if !envVarExists {
+		return "", false
+	}
+	launcherVerbosityLevelEnvVar, envVarExists := os.LookupEnv(services.ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY)
+	if !envVarExists {
+		return "", false
+	}
+
+	libvirtLogVerbosity, err := strconv.Atoi(launcherVerbosityLevelEnvVar)
+	if err != nil {
+		log.Log.Infof("cannot apply %s value %s - must be a number", services.ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY, launcherVerbosityLevelEnvVar)
+		return "", false
+	}
```

Comment:
One question about the logic.

IIUC, the way you've implemented it, setting custom log filters through the `kubevirt.io/libvirt-log-filters` annotation will only take effect if you have also enabled debug logs for libvirt with the `debugLogs` label and set the log verbosity with the `logVerbosity` label. Does it make sense to require all these knobs to be turned at once for anything to happen? I would expect things to work more like

* setting `kubevirt.io/libvirt-log-filters` enables those specific filters, regardless of other labels or annotations;
* setting `debugLogs` enables the default set of log filters, regardless of other labels or annotations;
* setting `logVerbosity` to a high enough number enables the default set of log filters, regardless of other labels or annotations.

Are my expectations or my understanding of the current implementation way off?

Note that IIUC at least the second case used to work exactly as I describe it above before your changes, but now it no longer does because `debugLogs` only takes effect if `logVerbosity` has been set to some integer value.

Reply:
Makes total sense.

I'm working on it + adding some more tests. Will repush soon